### PR TITLE
CSP headers

### PR DESCRIPTION
**Pull request by sleidig**
see issue: #950 

Adds Content-Security-Policy headers.

For deployment / migration:
- define `CSP_REPORT_URI` as environment var (get from sentry to send reports there https://aam-digital.sentry.io/settings/projects/aam-digital/security-headers/csp/ )
- optionally : define `CSP` specifying the full policy (necessary e.g. for dev server to allow *.aam-digital.net instead of *.aam-digital.com), start by copying the defalt from the Dockerfile.

### CODE REVIEW

File: build/README.md [hunk 1]
```diff
@@ -10,6 +10,14 @@ for every official (tagged) build.
 ## How to build & publish a new image
 You can simply create a new git tag and the CI setup will build and publish a docker image for that version.
 
+## Building locally
+Run the following commands from the root folder to build, run and kill the application on your local machine:
+```
+docker build -f build/Dockerfile -t aam/digital:latest .
+docker run -p=80:80 aam/digital:latest
```
Comment (TheSlimvReal) [inline build/README.md:17]:
```suggestion
docker run -p=80:80 --name aam-digital aam/digital:latest
```
so the `docker kill` command afterwards is working

File: build/Dockerfile [hunk 2]
```diff
@@ -62,6 +62,13 @@ ENV WEBDAV_URL="http://localhost"
 ENV COUCHDB_URL="http://localhost"
 # The language which should be loaded on default options are "en-US" and "de"
 ENV DEFAULT_LANGUAGE="en-US"
+
+# url to be reverse-proxied for logging endpoint (e.g. sentry.io)
+ENV LOGGING_URL=""
```
Comment (sleidig) [inline build/Dockerfile:67]: backends already use SENTRY endpoint env

**Comment by sleidig** [comment]
Two limitations that only have workarounds:
- https://github.com/angular/angular-cli/issues/20864 (workaround using https://stackoverflow.com/a/67582075/1473411 as I am not sure how to use the nonce feature of angular that the PRs are talking about)
- pouchdb requires unsafe-eval (https://github.com/pouchdb/pouchdb/issues/7853)

**Comment by sleidig** [comment]
deployed to https://dev.aam-digital.net for testing

File: build/Dockerfile [hunk 3]
```diff
@@ -73,6 +73,18 @@ ENV PORT=80
...
 ENV NOMINATIM_URL="https://nominatim.openstreetmap.org"
+# url to be reverse-proxied for logging endpoint (e.g. sentry.io)
+ENV LOGGING_URL=""
+
+# content security policy headers
+ENV CSP_REPORT_URI=""
+## additional allowed default sources, define URLs that you want to allow as CSP_SRC_ALLOW. This is only used if CSP_SRC is *not* overwritten
+ENV CSP_SRC_ALLOW="https://sentry.io https://api.github.com/repos/Aam-Digital/ https://matomo.aam-digital.org  https://*.aam-digital.com"
+## (optional) complete override of CSP using CSP_SRC; Warning: the defaults here will be overwritten but similar rules are required for smooth functioning of the app
+ENV CSP_SRC="default-src 'self' 'unsafe-eval' $CSP_SRC_ALLOW 'sha256-Sh1PNRUktjifFwuOicavxmlAcFpZMbUqQGiCIoKhDI8='; style-src 'self' 'unsafe-inline'; img-src 'self' data:; font-src 'self' data:"
```
Comment (sleidig) [inline build/Dockerfile:84]:
```suggestion
ENV CSP_SRC_ALLOW="https://matomo.aam-digital.org  https://*.aam-digital.com"
## (optional) complete override of CSP using CSP_SRC; Warning: the defaults here will be overwritten but similar rules are required for smooth functioning of the app
ENV CSP_SRC="default-src 'self' 'unsafe-eval' $CSP_SRC_ALLOW https://sentry.io https://api.github.com/repos/Aam-Digital/ https://a.tile.openstreetmap.org/ 'sha256-Sh1PNRUktjifFwuOicavxmlAcFpZMbUqQGiCIoKhDI8='; style-src 'self' 'unsafe-inline'; img-src 'self' data:; font-src 'self' data:"
```

**Review by TheSlimvReal — APPROVED** [review]
Seems to work well, lets see what will come in

**Pull request merged**